src/games/strategy/triplea/delegate/BattleTracker.java

```diff
@@ -1084,6 +1100,30 @@ public void sendBattleRecordsToGameData(final IDelegateBridge aBridge) {
     }
   }
 
+
+  void fightStrategicBombingRuns(final IDelegateBridge delegateBridge) {
+    boolean bombing = true;
+    fightStrategicBombingRuns(delegateBridge, () -> getPendingBattleSites(bombing),
+            (territory, battleType) -> getPendingBattle(territory, bombing, battleType));
+  }
+
+  @VisibleForTesting
+  void fightStrategicBombingRuns(final IDelegateBridge delegateBridge, Supplier<Collection<Territory>> pendingBattleSiteSupplier,
+             BiFunction<Territory, BattleType, IBattle> pendingBattleFunction) {
+    // Fight all air and bombing fightStrategicBombingRuns without needing user to click on them.
+    // The rules say that bombing battles should be done first, so automatically do those combats.
+    for( final Territory t : pendingBattleSiteSupplier.get()) {
+      final IBattle airRaid = pendingBattleFunction.apply(t, BattleType.AIR_RAID);
+      if( airRaid != null ) {
+        airRaid.fight(delegateBridge);
+      }
+      final IBattle bombingRaid = pendingBattleFunction.apply(t, BattleType.BOMBING_RAID);
+      if( bombingRaid != null ) {
+        bombingRaid.fight(delegateBridge);
+      }
```

Comment:
An Air Raid is the air battle preceding a bombing raid. In 1940 (optional in 1942.2) when one has a factory or base attacked, you have the option of sending up fighters to intercept the bombers. This defender decisions is (unfortunately) made after the attacker spawns the air raid. Theoretically, all air raid decisions should be made before any dice are rolled, although the consequences aren't too severe.

That's why I wanted to automate the air raids most of all. Group all the defender decisions together - scramble/kamikaze/intercept. It still won't be perfect because there is often an attacker decision about what facility will be attacked in 1940 (naval base/airbase/factory) so if there are two air raids in one turn there will still be a pass of control if the first one necessitates an attacker decision. However, it will resolve the 98% case of their only being one air raid.

Sorry if that is too long an answer.